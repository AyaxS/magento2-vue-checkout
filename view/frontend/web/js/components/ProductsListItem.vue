<template>
  <li class="product">
    <p class="products__title">
      {{ $t('productList.name') }}
      <strong>
        {{ product.name }}
      </strong>
    </p>

    <span class="product__price">
      {{ $t('productList.price') }}

      <strong>
        {{ product.price | currency }}
      </strong>
    </span>

    <div class="products__handler">
      <p class="products__information">
        {{ $t('productList.details') }}
      </p>

      <template v-for="detail in JSON.parse(product.options)">
        <span
          :key="detail.id"
          class="products__detail"
        >
          {{ detail.label }}:
          <strong>
            {{ detail.value }}
          </strong>
        </span>
      </template>

      <span class="products__detail">
        {{ $t('productList.qty') }}
        <strong>
          {{ product.qty }}
        </strong>
      </span>

      <template v-if="product.discount_percent > 0">
        <span class="product__detail">
          {{ $t('productList.dicsount') }} {{ product.discount_percent }}%
        </span>
      </template>
    </div>
  </li>
</template>

<script>
export default {
  props: {
    product: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
  .product {
    padding: 10px;
    border-bottom: 1px solid #c2c2c2;
    &:last-child {
      border-bottom: none;
    }
  }
</style>
