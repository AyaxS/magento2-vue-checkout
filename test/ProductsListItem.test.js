import { mount } from '@vue/test-utils'
import ProductsListItem from '../view/frontend/web/js/components/ProductsListItem.vue'

describe('ProductsListItem.test.js', () => {

  it('has received [] as the products property', () => {
    const wrapper = mount(ProductsListItem, {
      // Beaware that props is overriden using `propsData`
      propsData: {
        product: {
          base_discount_amount: '0.0000',
          base_price: '48.0000',
          base_price_incl_tax: '48.0000',
          base_row_total: '48.0000',
          base_row_total_incl_tax: '48.0000',
          base_tax_amount: '0.0000',
          discount_amount: '0.0000',
          discount_percent: '0.0000',
          item_id: '229',
          name: 'Hera Pullover Hoodie',
          options: '[{"value":"Blue","label":"Color"},{"value":"XS","label":"Size"}]',
          price: '48.0000',
          price_incl_tax: '48.0000',
          qty: 1,
          row_total: '48.0000',
          row_total_incl_tax: '48.0000',
          row_total_with_discount: '0.0000',
          tax_amount: '0.0000',
          tax_percent: '0.0000',
          weee_tax_applied: null,
          weee_tax_applied_amount: null
        }
      }
    })

    // expect(cmp.vm.products).toEqual([])
    // expect(wrapper.is('div')).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
})
