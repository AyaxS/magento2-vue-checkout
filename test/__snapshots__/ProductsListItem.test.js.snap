// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsListItem.test.js has the expected html structure 1`] = `
<div
  class="product"
>
  <p
    class="products__title"
  >
    
    Product name:
    
    <strong>
      
      Hera Pullover Hoodie
    
    </strong>
  </p>
   
  <span
    class="product__price"
  >
    
    Price:
    
    <strong>
      
      $48.00
    
    </strong>
  </span>
   
  <div
    class="products__handler"
  >
    <p
      class="products__information"
    >
      
      Product details:
    
    </p>
     
    <span
      class="products__detail"
    >
      
        Color:
        
      <strong>
        
          Blue
        
      </strong>
    </span>
    <span
      class="products__detail"
    >
      
        Size:
        
      <strong>
        
          XS
        
      </strong>
    </span>
     
    <span
      class="products__detail"
    >
      
      Qty:
      
      <strong>
        
        1
      
      </strong>
    </span>
     
    <!---->
  </div>
</div>
`;
